### ISSUE TITLE
Use official alpine instead of gliderlabs/alpine:3.1

### ISSUE THREAD

**Issue by develar** [mhart/alpine-node#2]
gliderlabs is now official docker library — https://registry.hub.docker.com/_/alpine/ Please use official.

**Comment by mhart** [comment]
Any reason for this? They both come from [the same source](https://github.com/gliderlabs/docker-alpine). And the `alpine` image doesn't have the `apk-install` helper – so it would make the Dockerfile more verbose.

**Comment by develar** [comment]
Because it is better to use common base image.

If you want to use java / node / custom image / data container / etc and you want to ensure that new node will be up in few seconds — you must reduce download size. And, so, you should select only one base image. If image A will use gliderlabs/alpine:3.1, but image B and C will use alpine:3.1 — functionality the same, but download time will be increased.

**Comment by mhart** [comment]
I'm not sure what you're talking about. Both `alpine` and `gliderlabs/alpine` are 5.033 MB

**Comment by mhart** [comment]
``` console
$ docker images | grep alpine
alpine                        latest          a1dd7097a8e8        2 weeks ago         5.033 MB
alpine                        edge            5b9d2dfd9d1a        2 weeks ago         5.316 MB
gliderlabs/alpine             edge            83f9dd5811b1        2 weeks ago         5.316 MB
gliderlabs/alpine             3.1             a8be93256def        2 weeks ago         5.033 MB
gliderlabs/alpine             latest          a8be93256def        2 weeks ago         5.033 MB
```

**Comment by mhart** [comment]
Unless there's some sort of official guidance from the gliderlabs team that everyone should stop using the `gliderlabs/alpine` images, I'm reticent to change anything here.

Their README still clearly references `gliderlabs/alpine`:

https://github.com/gliderlabs/docker-alpine

**Comment by mhart** [comment]
Also, `alpine` has less than half as many downloads as `gliderlabs/alpine` so chances are actually higher that people will have `gliderlabs/alpine` – not the other way around.

Happy to revisit if this situation changes or if there's some sort of official guidance.

**Comment by develar** [comment]
Thanks, I asked about it — https://github.com/gliderlabs/docker-alpine/issues/34

**Comment by mhart** [comment]
Cool, thanks!

**Comment by marcellodesales** [comment]
Also prefer official images... That guarantees that no 2 base images will be downloaded in my environment... It's not the size, but the duplication of base images doing the same thing...

**Comment by asafyish** [comment]
I would also prefer using the official image. just to be on the safe side.

**Comment by goloroden** [comment]
+1 for using an official image. It somewhat feels more "safe".

**Comment by josephpage** [comment]
+1 for using official alpine image

**Comment by stenote** [comment]
+1 for using official alpine image

**Comment by mhart** [comment]
OK, so there's finally been an update from the gliderlabs team on this: https://github.com/gliderlabs/docker-alpine/pull/65

So the two images are exactly the same, built from the same source by the same team, etc – except the `gliderlabs/alpine` image is built with the `-c` flag which adds the extra (tiny) `apk-install` script.

This script is quite useful for installing packages – and because we refer to it in our README when suggesting how to build images with native dependencies, removing it will break builds for anyone relying on the current tag structure.

However, now that Node.js is about to release 4.x, this could be a good time to move on if people so wish – we can switch to using the `alpine` image for Node.js 4.x and beyond and I'll update the documentation accordingly.

I'll close this issue once I have a build for Node.js 4.x

**Comment by mhart** [comment]
Just released 4.0.0 built on the official image. Closing this now, thanks for all the input!
